app/containers/ProductView/reducer.js

```diff
@@ -0,0 +1,34 @@
+/*
+ *
+ * ProductView reducer
+ *
+ */
+
+import { fromJS } from 'immutable';
+import {
+    GET_PRODUCT,
+    GET_PRODUCT_FAIL,
+    GET_PRODUCT_SUCCESS,
+} from './constants';
+
+export const initialState = fromJS({ data: {} });
+
+function productViewReducer(state = initialState, action) {
+    switch (action.type) {
+        case GET_PRODUCT:
+            return state
+                .set('loading', true);
+        case GET_PRODUCT_FAIL:
+            return state
+                .set('loading', false)
+                .set('error', 'Error occurs when get products details.');
```

Comment:
i suppose there is a standard message return from the API about what the error is, can we use that?